crypto/lhash/lhash.c

```diff
@@ -286,24 +323,32 @@ static OPENSSL_LH_NODE **getrn(OPENSSL_LHASH *lh,
 {
     OPENSSL_LH_NODE **ret, *n1;
     unsigned long hash, nn;
-    OPENSSL_LH_COMPFUNC cf;
 
-    hash = (*(lh->hash)) (data);
+    if (lh->hashw)
+        hash = lh->hashw(data, lh->hash);
+    else
+        hash = lh->hash(data);
```

Comment:
It's unclear why lh->hashw() shouldn't exist if lh->hash() does.

Reply:
I guess backwards compatibility?

Reply:
Either doall_util_fn() also needs to be changed to choice between the functions, or we need to say that the thunk functions are now required. I can't find any open source code that would have a problem with the requirement.

Reply:
Regarding the checking of hashw vs hash, yes, backwards compatibility was my concern - i.e. if a user compiles against openssl 3.1 and then runs with openssl 3.2, the DEFINE_LHASH_OF[_EX] wouldn't assign the thunking function, resulting in hashw being null.

Regarding doal_util_fn, that function need not be changed because the proper functions are selected in the calling points (OPENSSL_LH_doall / OPENSSL_LH_doall_arg / OPENSSL_LH_doall_arg_thunk).  I could refactor that if needed, but I wanted to minimize the code change